functions/Find-DbaAgentJob.ps1

```diff
@@ -246,7 +277,21 @@ Queries CMS server to return all SQL instances in the Production folder and then
                                 Write-Verbose "Finding job/s that have no email operator defined"
                                 $output += $jobs | Where-Object { $Category -contains $_.Category }
                             }
-
+                        IF ($Owner)
+                            {
+                                Write-Verbose "Finding job/s with owner critera"
+                                IF ($Owner -match "-")
+                                    {
+                                        $OwnerMatch = $Owner -replace "-", ""
+                                        Write-Verbose "Checking for jobs that NOT owned by: $OwnerMatch"
+                                        $output += $server.JobServer.jobs | Where-Object { $OwnerMatch -notcontains $_.OwnerLoginName } 
```

Comment:
I think that, instead of reassigning $jobs and $output so many times, building a where clause string ( as above with $dynstring) and only getting the $output once in the end, could improve the performance